src/lib/retry.ts

```diff
@@ -0,0 +1,211 @@
+// copied over from our internal lib
```

Comment:
This all looks pretty neat. I wonder if we should GPL it as a separate package 🤔 

https://github.com/Automattic/vip-go-node-internal/blob/3895523d8176ea1a9644aebd8acf9078c1126789/src/utils/retry/retry.ts

Reply:
Fun to think about! It's MIT now 😅 , but copyright is ours indeed if we plan to change the license. I honestly don't see much of a benefit of whichever license - it's a pretty common tool.